packages/core/src/providers/pinpoint/types/pinpoint.ts

```diff
@@ -34,22 +34,23 @@ export type PinpointAnalyticsEvent = {
 	metrics?: Record<string, number>;
 };
 
-export type PinpointUpdateEndpointParameters = PinpointServiceOptions & {
+// Common type that is required for operations that may trigger an endpoint update
+type PinpointCommonEndpointOptions = {
```

Comment:
Nit: To separate the "options" concept from the wrapping parameters
```suggestion
type PinpointCommonParameters = {
```